src/peer_select.cc

```diff
@@ -1152,9 +1157,8 @@ PeerSelector::interestedInitiator()
 
 bool
 PeerSelector::wantsMoreDestinations() const {
-    const auto maxCount = Config.forward_max_tries;
-    return maxCount >= 0 && foundPaths <
-           static_cast<std::make_unsigned<decltype(maxCount)>::type>(maxCount);
+    const auto maxCount = make_const_var_unsigned(Config.forward_max_tries);
+    return foundPaths < static_cast<size_t>(maxCount);
```

Comment:
Something went very wrong here:

1. Config.forward_max_tries is not a constant variable -- it should not be used with make_const_var_unsigned.
2. Config.forward_max_tries might even be negative -- it should not be casted to unsigned without checking.
3. The comparison of two unsigned variables should not need a cast.

Reply:
Missed, indeed.
IIRC what's happening here is that on some os/compiler combo we support , size_t is signed, hence the cast.
I'll iterate removing the whole maxCount construct; if it doesn't work, I'll see about adding some explicit checks but it may get hairy.